ddtrace/internal/ci_visibility/encoder.py

```diff
@@ -67,24 +73,57 @@ def _convert_span(span, dd_origin):
         # type: (Span, str) -> Dict[str, Any]
         sp = JSONEncoderV2._span_to_dict(span)
         sp = JSONEncoderV2._normalize_span(sp)
-        sp["type"] = span.span_type
+        if dd_origin is not None:
+            sp["meta"].update({"_dd.origin": dd_origin})
+        sp["type"] = span.get_tag(EVENT_TYPE) or span.span_type
         sp["duration"] = span.duration_ns
         sp["meta"] = dict(sorted(span._meta.items()))
         sp["metrics"] = dict(sorted(span._metrics.items()))
-        sp["trace_id"] = int(sp.get("trace_id") or "1")
-        sp["parent_id"] = int(sp.get("parent_id") or "1")
-        sp["span_id"] = int(sp.get("span_id") or "1")
-        sp["test_suite_id"] = 1  # TODO: populate with real ID
-        sp["test_session_id"] = 1  # TODO: populate with real ID
-        if COVERAGE_TAG_NAME in sp["meta"]:
-            del sp["meta"][COVERAGE_TAG_NAME]
```

Comment:
This condition needs to remain in the diff. It was based on a comment from @juan-fernandez the other day.

Reply:
I've moved this to `CIVisibilityEncoderV01._filter_ids()` to handle cleaning span meta items.